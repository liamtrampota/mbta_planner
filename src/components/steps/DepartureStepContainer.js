import React from 'react';
import '../../styles/StepContainer.css';

const DepartureStepContainer = ({ predictedDepartureSchedule }) => (
  <div>
    <h3>
      Upcoming departure times
    </h3>
    <div className="stepContainer">
      { predictedDepartureSchedule.map(departureTime => (
        <DepartureTime key={departureTime} departureTime={departureTime} />
      ))}
    </div>
  </div>
)

const DepartureTime = ({ departureTime }) => (
  <div style={{ border: '1px solid black', padding: '5px', margin: '5px', width: '100px' }}>
    {new Date(departureTime.attributes.departure_time).toLocaleTimeString()}
  </div>
)

export default DepartureStepContainer;
