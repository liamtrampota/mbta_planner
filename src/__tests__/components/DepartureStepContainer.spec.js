import React from 'react';
import { shallow } from 'enzyme';
import DepartureStepContainer from '../../components/steps/DepartureStepContainer';

const props = { predictedDepartureSchedule: [{ attributes: {departure_time: 'Sun Jan 24 2021 23:27:07 GMT-0500 (Eastern Standard Time)' }}]}

describe('DepartureStepContainer', () => {
  test('renders DepartureStepContainer', () => {
    const wrapper = shallow(<DepartureStepContainer {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
