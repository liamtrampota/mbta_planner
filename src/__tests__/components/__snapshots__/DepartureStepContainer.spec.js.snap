// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepartureStepContainer renders DepartureStepContainer 1`] = `ShallowWrapper {}`;
